packages/gestalt/src/Link.js

```diff
@@ -14,26 +20,83 @@ import useFocusVisible from './useFocusVisible.js';
 import layoutStyles from './Layout.css';
 
 type Props = {|
+  /**
+   * Supply a short, descriptive label for screen-readers to replace link texts that don't provide sufficient context about the link component behavior. Texts like `Click Here,` or `Read More` can be confusing when a screen reader reads them out of context. In those cases, we must pass an alternative text to replace the link text. It populates aria-label. Screen readers read the `accessibilityLabel` prop, if present, instead of the link text. See the [Accessibility](https://gestalt.pinterest.systems/link#accessibility) guidelines for more information.
```

Comment:
The link isn't correct: "Accessibility" needs to be capitalized to hit the correct anchor. (Also a nit: I would think to link "Accessibility guidelines", not just the first word)